lib/network/modules/components/shared/Label.js

```diff
@@ -178,141 +202,178 @@ class Label {
+   *  - multi-font `bold` is taken here as an example; the same applies to 
+   *    the other multi-fonts
+   *  - prototypes are totally ignored. The idea is to add option objects used as prototypes
+   *    to the pile, in the correct order.
    *
-   * @param {Object} options
-   * @param {Object} groupOptions
-   * @param {Object} defaultOptions
+   * ### Order:
+   *
+   *   - search in option group 'bold' in local properties
+   *   - search in main font option group in local properties
+   *
+   * ---------------------------------------------------------------------
+   * @param {Array.<object>} pile  sequence of option objects to consider.
+   *                               First item in list assumed to be the newly set options.
    */
-  propagateFonts(options, groupOptions, defaultOptions) {
-    if (!this.fontOptions.multi) return;
+  propagateFonts(pile) {
+    let fontPile = [];   // sequence of font objects to consider, order important
+    let mods = ['bold', 'ital', 'boldital', 'mono'];
+
 
     /**
-     * Resolve the font options path.
-     * If valid, return a reference to the object in question.
-     * Otherwise, just return null.
+     * Add given font option object to the list of objects (the 'pile') to consider for determining
+     * multi-font option values.
      *
-     * @param {Object} options base object to determine path from
-     * @param {'bold'|'ital'|'boldital'|'mono'|'normal'} [mod=undefined] if present, sub path for the mod-font
-     * @returns {Object|null}
+     * @param {object}  options instance to add to pile
+     * @param {boolean} [isFontOption=false] if true, use option directly instead of accessing member var 'font'
      */
-    var pathP = function(options, mod) {
-       if (!options || !options.font) return null;
+    let addFontToPile = (options, isFontOption = false) => {
+      if (options === undefined) return;
 
-      var opt = options.font;
-
-      if (mod) {
-        if (!opt[mod]) return null;
-        opt = opt[mod];
+      let item;
+      if (isFontOption) {
+        item = options;
+      } else {
+        if (options.font === undefined || options.font === null) return;
+        item = options.font;
       }
 
-      return opt;
+      fontPile.push(item);
     };
 
 
-    /**
-     * Get property value from options.font[mod][property] if present.
-     * If mod not passed, use property value from options.font[property].
-     *
-     * @param {Label.options} options
-     * @param {'bold'|'ital'|'boldital'|'mono'|'normal'} mod
-     * @param {string} property
-     * @return {*|null} value if found, null otherwise.
-     */
-    var getP = function(options, mod, property) {
-      let opt = pathP(options, mod);
+   /**
+    * Collect all own-property values from the font pile that aren't multi-font option objectss.
+    *
+    * @returns {object} object with all current own basic font properties
+    */
+    let getBasicOptions = () => {
```

Comment:
I'm seeing a lot of nested function calls and wondering if it is a micro optimization to bring them up a level, so that they're initialized with the class, rather than initialized with every call to the function that encapsulates them.

Reply:
OK. My long-term planning is to isolate this pile handling code in a separate class. This is the approach that I want to use to replace the prototype chaining in options. I thought it was too early to make a class of it now.

But fair enough, I can define them as private methods for the time being. Wrt. 'micro optimization', it's an easy task to raise them to class-level, and every teeny bit helps.